### PR TITLE
Rename every instance of `caret_blink_speed` to `caret_blink_interval`

### PR DESCRIPTION
**Pull request by Mickeon**
As brought up in https://github.com/godotengine/godot/issues/54161#issuecomment-1175135591 and https://github.com/godotengine/godot/issues/54161#issuecomment-1175247779.

~This PR was renaming it to caret_blink_rate~
~While none of the comments mentioned "rate", I chose it because the Windows settings and registry call it **CursorBlinkRate**. I was about to rename it to `caret_blink_time`, but just as I was creating this Pull Request, I realised how much better `caret_blink_rate` rolls off the tongue, especially because it can't be confused for the internal `caret_blink_timer` variable.~

It's been changed in the **editor settings**, and in the **LineEdit** and **TextEdit** classes.
Affects setters and getters, and passed parameters, too.

### CODE REVIEW

**Comment by Mickeon** [comment]
Speaking of which, the **LineEdit** and **TextEdit** description, while short, has also been a bit misleading?
> Duration (in seconds) of a caret's blinking cycle.

"cycle" implies the time it takes for both states (visible and hidden) to complete, but the current behaviour simply toggles between the two every `caret_blink_speed` seconds, so it's just half the "total" caret blinking cycle.

**Review by akien-mga — APPROVED** [review]
Approved in PR meeting. Needs rebase.

**Review by aaronfranke — APPROVED** [review]
The old name of "speed" was definitely wrong, since an increased value is slower.

**Comment by akien-mga** [comment]
So I got confused in the meeting a bit but "rate" is often used as a synonym to "frequency", which is the inverse of the time period.

Maybe "caret_blink_interval" would be better?

**Comment by Mickeon** [comment]
I wanted to keep it shorter but both _rate_ and _interval_ make more sense than _speed_. Sounds good to me.

**Review by aaronfranke — APPROVED** [review]

**Comment by Mickeon** [comment]
Wait! Forgot C# renames

File: scene/gui/line_edit.cpp [hunk 1]
```diff
@@ -1392,13 +1392,13 @@ void LineEdit::set_caret_force_displayed(const bool p_enabled) {
...
+float LineEdit::get_caret_blink_interval() const {
+	return caret_blink_interval;
 }
 
-void LineEdit::set_caret_blink_speed(const float p_speed) {
-	ERR_FAIL_COND(p_speed <= 0);
-	caret_blink_speed = p_speed;
+void LineEdit::set_caret_blink_interval(const float p_rate) {
+	ERR_FAIL_COND(p_rate <= 0);
+	caret_blink_interval = p_rate;
```
Comment (aaronfranke) [inline scene/gui/line_edit.cpp:1401]:
```suggestion
void LineEdit::set_caret_blink_interval(const float p_interval) {
	ERR_FAIL_COND(p_interval <= 0);
	caret_blink_interval = p_interval;
```

**Review by aaronfranke — CHANGES_REQUESTED** [review]

File: scene/gui/line_edit.cpp (hunk 1 above)
Comment (Mickeon) [reply scene/gui/line_edit.cpp]: Fixed every occurrence.

**Review by aaronfranke — APPROVED** [review]

File: editor/project_converter_3_to_4.cpp [hunk 2]
```diff
@@ -278,6 +279,7 @@ static const char *gdscript_function_renames[][2] = {
 	{ "get_camera", "get_camera_3d" }, // Viewport -> this is also convertable to get_camera_2d, broke GLTFNode
 	{ "get_cancel", "get_cancel_button" }, // ConfirmationDialog
 	{ "get_caption", "_get_caption" }, // AnimationNode
+	{ "get_caret_blink_speed", "get_caret_blink_interval" }, // TextEdit, LineEdit
```
Comment (akien-mga) [inline editor/project_converter_3_to_4.cpp:282]: This method name didn't exist in `3.x` so it doesn't need to be handled. Same for `set_caret_blink_speed`.

File: editor/project_converter_3_to_4.cpp [hunk 3]
```diff
@@ -700,6 +703,7 @@ static const char *csharp_function_renames[][2] = {
 	{ "GetCamera", "GetCamera3d" }, // Viewport -> this is also convertable to getCamera2d, broke GLTFNode
 	{ "GetCancel", "GetCancelButton" }, // ConfirmationDialog
 	{ "GetCaption", "_GetCaption" }, // AnimationNode
+	{ "GetCaretBlinkSpeed", "GetCaretBlinkInterval" }, // TextEdit, LineEdit
```
Comment (akien-mga) [inline editor/project_converter_3_to_4.cpp:706]: This wasn't the `3.x` name of these methods. It's `Cursor{Set,Get}BlinkSpeed` that should be changed.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**